### PR TITLE
[abstracts] code style changes for abstract unify functions

### PR DESCRIPTION
**Pull request by vonagam**
So this is style changes separated from previous PR.

Some changes in no particular order:

- `Abstract.find_to_from` returns, just like `find_to` and `find_from`, but it does also include selected `ab` and `tl`.

- Order of definition for abstract function with `_to` and `_from` is `_from` then `_to`. (`to` cast has priority over `from`, but `from-to` reads better than `to-from`, so 50-50, but in unification it is in this order, so `from-to` wins)

- Arguments for all abstract unify functions start with `uctx a b ab tl`. (In `Abstract.find_to` and `Abstract.find_from` `a` and `b` are omitted respectively, since places where they are called do not have those.)

- `unify_` for error throwing void functions and `unifies_` for boolean returning functions.

- `tl` instead of `pl` (in `Abstract` it was `pl`, in `tUnification` `tl`, searched code base - `tl` is more prevalent, so went with it).

- Only real non-cosmetic change - check for recursive cast size.

### CODE REVIEW

File: src/core/tUnification.ml [hunk 1]
```diff
@@ -883,54 +873,71 @@ and unify_anons uctx a b a1 a2 =
...
-				let map = apply_params ab.a_params tl in
-				let monos = Monomorph.spawn_constrained_monos map cf.cf_params in
-				let map t = map (apply_params cf.cf_params monos t) in
-				let athis = map ab.a_this in
-				(* we cannot allow implicit casts when the this type is not completely known yet *)
-				with_variance uctx (type_eq {uctx with equality_kind = EqStrict}) athis (map ta);
-				unify_func (map t) b;
-			| _ -> die "" __LOC__)
+and rec_stack_abstract_unifies a b frun =
+	List.length abstract_cast_stack.rec_stack < 10
```
Comment (Simn) [inline src/core/tUnification.ml]: Doing something like this seems wrong. If there's a problem with recursion we should look into why exactly that happens and then fix it. I don't think putting arbitrary stack depths here is the right thing to do.

Comment (vonagam) [reply src/core/tUnification.ml]: In dummy example of:

```haxe
abstract Foo<T>(T) from T to Foo<Null<T>> {}

class Main {
  static public function main() {
    var x: Foo<Int> = 0;
    var y: Int = x;
  }
}
```

It just goes deeper into `Foo<Null<Null<...>>>`, as it is not equal to previous types in recursion stack it keeps going forever. Any ideas about how can this be fixed better?

Comment (Simn) [reply src/core/tUnification.ml]: This should be in the same category as the various problems with recursive `typedef`s that we used to have.

Although the first question is if something like this should even be allowed. We don't allow `typedef C<T> = C<Null<T>>` either.

This is a separate issue though, so I'd say let's just hold back on this change for now. Or do your changes expose this particular problem?

Comment (vonagam) [reply src/core/tUnification.ml]: Hold back it is. Done.

**Pull request merged**